File: docs/guides/java/src/test/java/com/algolia/saveImageClassifications.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

class saveImageClassifications {

  static class Image {

    public String imageURL;
    public String objectID;
    public List<Map<String, Object>> objects;

    public Image(String imageURL, String objectID, List<Map<String, Object>> objects) {
      this.imageURL = imageURL;
      this.objectID = objectID;
      this.objects = objects;
    }
  }

  // Retrieve labels
  static Image getImageLabels(String imageURL, String objectID, float scoreLimit) {
    // Implement your image classification logic here
    return new Image(imageURL, objectID, new ArrayList<>());
  }

  public static void main(String[] args) throws Exception {
    // API key ACL should include editSettings / addObject
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")) {
      var hits = client.browseObjects("<YOUR_INDEX_NAME>", Image.class);

      List<Image> records = new ArrayList<>();
      for (var hit : hits) {
        var image = getImageLabels(hit.imageURL, hit.objectID, 0.5f);
        records.add(image);
      }

      // Update records with image classifications
      client.partialUpdateObjects("<YOUR_INDEX_NAME>", records, true);
    } catch (Exception e) {
      System.out.println(e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/saveImageClassifications.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.BrowseParamsObject
import com.algolia.client.transport.*
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonObject

suspend fun saveImageClassifications() {
  data class Image(val imageURL: String, val objectID: String, val objects: List<Map<String, Any>>)

  // Retrieve labels
  fun getImageLabels(imageURL: String, objectID: String, scoreLimit: Float): Image {
    // Implement your image classification logic here
    return Image(imageURL, objectID, emptyList())
  }

  // API key ACL should include editSettings / addObject
  try {
    val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

    val images: MutableList<Image> = mutableListOf()

    client.browseObjects(
      indexName = "<YOUR_INDEX_NAME>",
      params = BrowseParamsObject(),
      aggregator = { browseResponse ->
        images.addAll(
          browseResponse.hits.map {
            val props = it.additionalProperties ?: emptyMap()
            return@map getImageLabels(
              props.getOrElse("imageURL") { "" }.toString(),
              it.objectID,
              0.5f,
            )
          }
        )
      },
    )

    val records = images.map { Json.encodeToJsonElement(it).jsonObject }

    // Update records with image classifications
    client.partialUpdateObjects(
      indexName = "<YOUR_INDEX_NAME>",
      objects = records,
      createIfNotExists = true,
    )
  } catch (e: Exception) {
    println(e.message)
  }
}
